plugins/modules/aci_access_port_block_to_access_port.py

```diff
@@ -264,43 +351,46 @@
 def main():
     argument_spec = aci_argument_spec()
     argument_spec.update(
-        leaf_interface_profile=dict(type='str', aliases=['leaf_interface_profile_name']),  # Not required for querying all objects
+        interface_profile=dict(type='str', aliases=['leaf_interface_profile_name', 'leaf_interface_profile']),  # Not required for querying all objects
         access_port_selector=dict(type='str', aliases=['name', 'access_port_selector_name']),  # Not required for querying all objects
-        leaf_port_blk=dict(type='str', aliases=['leaf_port_blk_name']),  # Not required for querying all objects
-        leaf_port_blk_description=dict(type='str'),
+        port_blk=dict(type='str', aliases=['leaf_port_blk_name', 'leaf_port_blk']),  # Not required for querying all objects
+        port_blk_description=dict(type='str', aliases=['leaf_port_blk_description']),
         from_port=dict(type='str', aliases=['from', 'fromPort', 'from_port_range']),
         to_port=dict(type='str', aliases=['to', 'toPort', 'to_port_range']),
         from_card=dict(type='str', aliases=['from_card_range']),
         to_card=dict(type='str', aliases=['to_card_range']),
         state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        type=dict(type='str', default='leaf', choices=['fex', 'leaf']),   # This parameter is not required for querying all objects
     )
 
     module = AnsibleModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
         required_if=[
-            ['state', 'absent', ['access_port_selector', 'leaf_port_blk', 'leaf_interface_profile']],
-            ['state', 'present', ['access_port_selector', 'leaf_port_blk', 'from_port', 'to_port', 'leaf_interface_profile']],
+            ['state', 'absent', ['access_port_selector', 'port_blk', 'interface_profile']],
+            ['state', 'present', ['access_port_selector', 'port_blk', 'from_port', 'to_port', 'interface_profile']],
         ],
     )
 
-    leaf_interface_profile = module.params.get('leaf_interface_profile')
+    interface_profile = module.params.get('interface_profile')
     access_port_selector = module.params.get('access_port_selector')
-    leaf_port_blk = module.params.get('leaf_port_blk')
-    leaf_port_blk_description = module.params.get('leaf_port_blk_description')
+    port_blk = module.params.get('port_blk')
+    port_blk_description = module.params.get('port_blk_description')
     from_port = module.params.get('from_port')
     to_port = module.params.get('to_port')
     from_card = module.params.get('from_card')
     to_card = module.params.get('to_card')
     state = module.params.get('state')
+    type_port = module.params.get('type')
 
     aci = ACIModule(module)
+    class_rn = 'infraFexP/fexprof' if type_port == 'fex' else 'infraAccPortP/accportprof'
```

Comment:
I would define the aci_class and the aci_rn so that you don't have to split it just after.